matrix/server.py

```diff
@@ -1618,6 +1618,8 @@ def handle_response(self, response):
 
         if isinstance(response, ErrorResponse):
             self.handle_error_response(response)
+            room_buffer = self.room_buffers[response.room_id]
```

Comment:
Not every `ErrorResponse` has a `room_id` attribute, you'll have to add a more specific `isinstance` check checking for `RoomMessagesError` would make the most sense I would say.